src/basic/util.h

```diff
@@ -148,6 +148,13 @@ static inline int memcmp_safe(const void *s1, const void *s2, size_t n) {
         return memcmp(s1, s2, n);
 }
 
+/* Only for small arrays! Does double evaluation of the argument. */
+#define array_is_zero(arr)                                      \
+        ({                                                      \
+                __typeof__(arr) _zzz = {};                      \
+                memcmp((arr), _zzz, sizeof(_zzz)) == 0;         \
+        })
```

Comment:
hmm, wouldn't it be better to add memory_all_zeroes() or so as regular function that just iterates through the thing bytewise and checks for non-zero? Sure, would not be as fast, but at least work with all kinds of memory?

Reply:
maybe copy something like this into our sources somewhere: 

https://github.com/rustyrussell/ccan/blob/master/ccan/mem/mem.c#L92